<Type Name="UpdateSourceExceptionFilterCallback" FullName="System.Windows.Data.UpdateSourceExceptionFilterCallback">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="65b5bfbadcdb56e0bc2f5373d8d94483568f6634" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="MT" />
    <Meta Name="ms.contentlocale" Value="pl-PL" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
    <Meta Name="ms.locfileid" Value="30699185" />
  </Metadata>
  <TypeSignature Language="C#" Value="public delegate object UpdateSourceExceptionFilterCallback(object bindExpression, Exception exception);" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed UpdateSourceExceptionFilterCallback extends System.MulticastDelegate" />
  <TypeSignature Language="DocId" Value="T:System.Windows.Data.UpdateSourceExceptionFilterCallback" />
  <TypeSignature Language="VB.NET" Value="Public Delegate Function UpdateSourceExceptionFilterCallback(bindExpression As Object, exception As Exception) As Object " />
  <TypeSignature Language="C++ CLI" Value="public delegate System::Object ^ UpdateSourceExceptionFilterCallback(System::Object ^ bindExpression, Exception ^ exception);" />
  <AssemblyInfo>
    <AssemblyName>PresentationFramework</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Delegate</BaseTypeName>
  </Base>
  <Parameters>
    <Parameter Name="bindExpression" Type="System.Object" />
    <Parameter Name="exception" Type="System.Exception" />
  </Parameters>
  <ReturnValue>
    <ReturnType>System.Object</ReturnType>
  </ReturnValue>
  <Docs>
    <param name="bindExpression">Obiekt wyjątku.</param>
    <param name="exception">Napotkano wyjątek.</param>
    <summary>Reprezentuje metodę, która obsługuje wyjątki, które są generowane podczas aktualizacji wiążąca wartość źródła. Ten musi być używany z <see cref="T:System.Windows.Controls.ExceptionValidationRule" />.</summary>
    <returns>Obiekt, który zazwyczaj jest jedną z następujących czynności:  
  
 <list type="table"><listheader><term> Wartość  
  
 </term><description> Opis elementu  
  
 </description></listheader><item><term><see langword="null" /></term><description> Aby zignorować wszelkie wyjątki. Domyślne zachowanie (jeśli istnieje nie <see cref="T:System.Windows.Data.UpdateSourceExceptionFilterCallback" />) jest utworzenie <see cref="T:System.Windows.Controls.ValidationError" /> z wyjątkiem i dodaje go do <see cref="P:System.Windows.Controls.Validation.Errors" /> kolekcji elementu powiązania.  
  
 </description></item><item><term> Dowolny obiekt  
  
 </term><description> Aby utworzyć <see cref="T:System.Windows.Controls.ValidationError" /> obiekt z <see cref="P:System.Windows.Controls.ValidationError.ErrorContent" /> wartość tego obiektu.  
  
 <see cref="T:System.Windows.Controls.ValidationError" /> Obiekt jest dodawany do <see cref="P:System.Windows.Controls.Validation.Errors" /> kolekcji elementu powiązania.  
  
 </description></item><item><term> A <see cref="T:System.Windows.Controls.ValidationError" /> obiektu  
  
 </term><description> Aby ustawić <see cref="T:System.Windows.Data.BindingExpression" /> lub <see cref="T:System.Windows.Data.MultiBindingExpression" /> obiekt jako <see cref="P:System.Windows.Controls.ValidationError.BindingInError" />. <see cref="T:System.Windows.Controls.ValidationError" /> Obiekt jest dodawany do <see cref="P:System.Windows.Controls.Validation.Errors" /> kolekcji elementu powiązania.  
  
 </description></item></list></returns>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 Jeśli został skojarzony <xref:System.Windows.Controls.ExceptionValidationRule> z Twojej <xref:System.Windows.Data.Binding> obiektu mieć możliwość użycia <xref:System.Windows.Data.Binding.UpdateSourceExceptionFilter%2A> właściwości można ustawić tego wywołania zwrotnego zapewnienie niestandardowej logiki do obsługi wyjątków. To wywołanie zwrotne jest wywoływany napotkaniu wyjątku podczas aparat wiązania aktualizacji wiążąca wartość źródła.  
  
 Jeśli <xref:System.Windows.Data.Binding.UpdateSourceExceptionFilter%2A> nie jest określona w <xref:System.Windows.Data.Binding>, tworzy aparat wiązania <xref:System.Windows.Controls.ValidationError> z wyjątkiem i dodaje go do <xref:System.Windows.Controls.Validation.Errors%2A?displayProperty=nameWithType> kolekcji elementu powiązania.  
  
   
  
## Examples  
 <xref:System.Windows.Controls.TextBox.Text%2A> Właściwości następujące <xref:System.Windows.Controls.TextBox> jest powiązany z danymi właściwości source `Age3` jest typu `int`. <xref:System.Windows.Controls.ExceptionValidationRule> Sprawdza, czy wyjątki, które są generowane podczas aktualizacji właściwości source (na przykład gdy użytkownik wprowadzi wartość, której nie można przekonwertować na liczbę całkowitą).  
  
 [!code-xaml[BindValidation#tbExceptionFilter](~/samples/snippets/csharp/VS_Snippets_Wpf/BindValidation/CSharp/Window1.xaml#tbexceptionfilter)]  
  
 Masz opcję, aby zapewnić niestandardowej logiki do obsługi tych wyjątków. Poniższy przykład przedstawia użycie <xref:System.Windows.Data.Binding.UpdateSourceExceptionFilter%2A> właściwości można ustawić <xref:System.Windows.Data.UpdateSourceExceptionFilterCallback>:  
  
 [!code-csharp[BindValidation#filtercallback](~/samples/snippets/csharp/VS_Snippets_Wpf/BindValidation/CSharp/Window1.xaml.cs#filtercallback)]  
  
 Poniżej przedstawiono przykład stosowania <xref:System.Windows.Data.UpdateSourceExceptionFilterCallback>:  
  
 [!code-csharp[BindValidation#Handler](~/samples/snippets/csharp/VS_Snippets_Wpf/BindValidation/CSharp/Window1.xaml.cs#handler)]  
  
 Pełny przykład, zobacz [powiązania przykładowych weryfikacji](http://go.microsoft.com/fwlink/?LinkID=159972).  
  
 ]]></format>
    </remarks>
    <altmember cref="P:System.Windows.Data.MultiBinding.UpdateSourceExceptionFilter" />
  </Docs>
</Type>